components/toast/toast.jsx

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import UnreadBelowIcon from 'components/widgets/icons/unread_below_icon';
+import CloseIcon from 'components/widgets/icons/close_icon';
+
+function empty() {
+    // do nothing
+}
+
+const someBigNum = 1000;
+
+export default class Toast extends React.PureComponent {
+    static propTypes = {
+        jumpTo: PropTypes.func.isRequired, // required?
+        jumpToMessage: PropTypes.string,
+        jumpFadeOutDelay: PropTypes.number,
+        onDismiss: PropTypes.func,
+        order: PropTypes.number,
+        children: PropTypes.element,
+        show: PropTypes.bool.isRequired,
+        extraClasses: PropTypes.string,
+        showOnlyOnce: PropTypes.bool.isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+
+            //once hide is set to true, only mounting again from the caller will show it again. This way, once we have used one button the toast doesn't show up again.
+            // if it should only be shown once, we set it to the value of show.
+            hide: this.props.showOnlyOnce ? !this.props.show : false,
+        };
+    }
+
+    shouldNeverShowAgain = () => {
+        // if it should never be seen again, hide it
+        if (this.props.showOnlyOnce) {
+            this.setState({hide: true});
+        }
+    }
+
+    handleJump = () => {
+        this.props.jumpTo();
+        setTimeout(() => this.shouldNeverShowAgain(), this.props.jumpFadeOutDelay);
+
+        // TODO: telemetry
+    }
+
+    handleDismiss = () => {
+        this.props.onDismiss();
+        this.shouldNeverShowAgain();
+
+        // TODO: add telemetry
+    }
+
+    render() {
+        let classes = `toast ${this.props.extraClasses}`;
+        if (this.props.show && !this.state.hide) {
+            classes += ' toast__visible';
+        }
+
+        return (
+            <div
+                className={classes}
+                style={{zIndex: this.props.order}}
```

Comment:
If we're just stacking the Toasts, is there any possibility we're going to get some weird visual effects from that like stacking drop shadows increasing in opacity?

Reply:
I was planning on testing this, as the only ones that might be one over the other is the permalink one and the new messages arrived one. Although I'm more concerned on translucency of the toast than on the shadows

Reply:
If it causes issues, maybe instead of multiple Toasts with one message each, we could use one Toast with multiple messages, and it shows the first relevant one? 0/5 on how it works specifically